dali/c_api/c_api.cc

```diff
@@ -130,6 +130,27 @@ int64_t* daliShapeAt(daliPipelineHandle* pipe_handle, int n) {
   return daliShapeAtTypedHelper(pipe_handle, n, -1);
 }
 
+template <typename T>
+static dali_data_type_t daliTypeAtHelper(dali::DeviceWorkspace* ws, int n) {
```

Comment:
Why not use a reference for ws?

Reply:
ctrl+c; ctrl+v probably.

Reply:
Oh, I know. It's because we store the pointer to a ws in pipe_handle. So I left the pointer instead of dereferencing.